Fix/improve layout

Rhian opened the pull request:
This is a fork of https://github.com/Think-and-Dev/solidoc which is a documentation solution for smart contracts. 

The changes in this PR update the module to document contracts up to v0. 8.14 of Solidity. In addition to providing support for documenting structs, it also includes the following updates:

- Better support for Events, which were intermittently missed before
- Enhanced presentation of return values (now tabulated instead of free text)
- Improved presentation of Constructor documentation

To test the output, follow the instructions in this draft PR: https://github.com/bosonprotocol/boson-protocol-contracts/pull/57

Once this PR is approved and merged, I will update the version in the `package.json`, add an explainer line in the `README` and push to the Boson registry.

cliffhall reviewed (commented):
* We should add a `.gitignore `with `.DS_Store` in it.

zajck commented:
I found some layout problems with [structs](https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-BosonTypes/#structs)

![image](https://user-images.githubusercontent.com/64400885/193756658-34d1cd9f-7485-4684-be54-0fa9913e5869.png)

1. First struct field should also be indented.
2. Solidity uses the `dataType variableName` style, not the other way around as it is now. I personally don't care that much, but I think it would be easier for other developers if we used the solidity style.
3. When enum types are part of a struct, they include the word "enum", which again is not how it's done in the solidity (where enum is omitted). I think that that should be removed for consistency.

The example from the picture above would then look like this:
```
struct AuthToken {
    uint256 tokenId;
    BosonTypes.AuthTokenType tokenType;
}
```

cliffhall commented:
<img width="457" alt="Screen Shot 2022-10-04 at 3 27 19 PM" src="https://user-images.githubusercontent.com/871933/193930461-60695369-4768-4af1-a5f5-a26029f70312.png">

Another small struct issue, extra space before the commas.

cliffhall reviewed (dismissed):
LGTM! 👍

Rhian commented:
> Just noticed, you will also need to change the package name in package.json. It's currently `@thinkanddev/solidoc`. I'm sure you'll run into it. I suppose `@bosonprotocol/solidoc`?

Fixed

Rhian commented:
> * We should add a `.gitignore `with `.DS_Store` in it.

Fixed in `0d9deda1891bd97667f37689cdc8a64e3f62378c`

Rhian commented:
> <img alt="Screen Shot 2022-10-04 at 3 27 19 PM" width="457" src="https://user-images.githubusercontent.com/871933/193930461-60695369-4768-4af1-a5f5-a26029f70312.png">
> 
> Another small struct issue, extra space before the commas.

Fixed - please validate here: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-BosonTypes

Rhian commented:
> I found some layout problems with [structs](https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-BosonTypes/#structs)
> 
> ![image](https://user-images.githubusercontent.com/64400885/193756658-34d1cd9f-7485-4684-be54-0fa9913e5869.png)
> 
> 1. First struct field should also be indented.
> 2. Solidity uses the `dataType variableName` style, not the other way around as it is now. I personally don't care that much, but I think it would be easier for other developers if we used the solidity style.
> 3. When enum types are part of a struct, they include the word "enum", which again is not how it's done in the solidity (where enum is omitted). I think that that should be removed for consistency.
> 
> The example from the picture above would then look like this:
> 
> ```
> struct AuthToken {
>     uint256 tokenId;
>     BosonTypes.AuthTokenType tokenType;
> }
> ```

Fixed - please validate here: Fixed - please validate here: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-BosonTypes

Rhian commented on builders/argument-builder.js:
```diff
@@ -25,11 +25,11 @@ module.exports = {
 
-      description.includes(" - ") ?
+      description.includes("-") ?
         description.replace(" - ", "").trim() :
```
Fixed.

Rhian commented:
> I read other comments and I think these two were not reported yet:
> 
> 1. unnamed return areguments are not proper rendered. A lot of examples in `ConfigHandlerFacet`
>    Code:
>    ![image](https://user-images.githubusercontent.com/64400885/188638057-002e1871-e2ce-476d-b1f8-15dc6238f071.png)
>    Documentation: ([Link to the example](https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-ConfigHandlerFacet#gettokenaddress))
>    ![image](https://user-images.githubusercontent.com/64400885/188638319-b7a121df-d8b8-4d47-b4a0-eab834d8760c.png)
> 2. Dashes in the description of parameters are not removed. We use dash between variable name and the description both in return arguments and params. For returns dash is removed, while for param it is not.
>    [Link to the example](https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-IBosonGroupHandler#getgroup)
>    ![image](https://user-images.githubusercontent.com/64400885/188639355-5d4f03b3-3855-459d-9ec6-060f6d2c536f.png)

Fixed - please validate here: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-IBosonConfigHandler#gettokenaddress

Rhian commented:
> At the top of the contract files we have a comment with an @title tag and an @notice tag. For some contracts, the @notice text is rendered (e.g. https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-FundsHandlerFacet/ and https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-DisputeHandlerFacet), but for most it is not (e.g. https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-ExchangeHandlerFacet and https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-AgentHandlerFacet)

Fixed in `3c8d0791d0a31d12ae0d5d92848b14025b18207e`

Rhian commented:
> Adding this comment from slack: I noticed that the description for modifier arguments is not being rendered, and the type between the brackets isn’t linked to the type page like it is for function arguments. See https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/contracts/version-undefined-BeaconClientBase/ onlyRole modifier

We no longer have this file on the website, but I've fixed and tested locally, as shown here. 

<img width="809" alt="modifiers" src="https://user-images.githubusercontent.com/1880450/195567000-7ed53066-15d1-48d7-aa1e-313a6dd66963.png">

Rhian commented:
> ![getAgent code](https://user-images.githubusercontent.com/42716568/187227955-bdbc3627-3a32-4076-baed-b4d24eb72649.png) ![getAgent Documentation](https://user-images.githubusercontent.com/42716568/187227963-c632f707-caf6-4b13-b448-763c265f42fe.png) I noticed the Description is missing for arguments. I do see the Description for return values.

Please validate here: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-IBosonDisputeHandler

Rhian commented:
Fixed in `bc9414f87e72d876d606c5cef63681f7008e47f0`

cliffhall commented:
<img width="481" alt="Screen Shot 2022-10-13 at 9 21 10 AM" src="https://user-images.githubusercontent.com/871933/195608882-68c7c2d5-e683-4f0d-a020-a8c696774b1f.png">

I see the "space before comma" fix worked, but just noticed that, like enums previously, when included in a struct, another struct has the word 'stuct' brefore the type that should be removed.

cliffhall commented:
<img width="536" alt="Screen Shot 2022-10-13 at 9 27 39 AM" src="https://user-images.githubusercontent.com/871933/195609692-55c47bf1-9c4b-4f39-b5ee-84220eb5ed1b.png">

I see that the "description missing from arguments table" is fixed, but then noticed this one weird instance, the name and descriptions are placeholders or something, but they have actual text in the natspec.

Also descriptions are not shown for return values that are not structs, i.e., normal types
<img width="569" alt="Screen Shot 2022-10-13 at 9 29 42 AM" src="https://user-images.githubusercontent.com/871933/195611532-a0049ced-f3d1-4761-9ff1-25cc1c7ae4b2.png">

Rhian commented:
> <img alt="Screen Shot 2022-10-13 at 9 27 39 AM" width="536" src="https://user-images.githubusercontent.com/871933/195609692-55c47bf1-9c4b-4f39-b5ee-84220eb5ed1b.png">
> 
> I see that the "description missing from arguments table" is fixed, but then noticed this one weird instance, the name and descriptions are placeholders or something, but they have actual text in the natspec.
> 
> Also descriptions are not shown for return values that are not structs, i.e., normal types <img alt="Screen Shot 2022-10-13 at 9 29 42 AM" width="569" src="https://user-images.githubusercontent.com/871933/195611532-a0049ced-f3d1-4761-9ff1-25cc1c7ae4b2.png">

Please re-review: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-IBosonDisputeHandler#getdispute

cliffhall reviewed (approved):
LGTM! 👍
